### PR TITLE
docs(category): add some api docs

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/category/src/models/category-breadcrumb.ts
Comment (xelaint) [inline libs/category/src/models/category-breadcrumb.ts:19]: is `trie` supposed to be `tree`?

File: libs/category/src/models/category-breadcrumb.ts
Comment (xelaint) [inline libs/category/src/models/category-breadcrumb.ts:20]: is `trie` supposed to be `tree`?

File: libs/category/src/models/generic-category.ts
Comment (xelaint) [inline libs/category/src/models/generic-category.ts:27]: `the category`?

File: libs/category/src/models/generic-category.ts
Comment (xelaint) [inline libs/category/src/models/generic-category.ts:31]: `the category`?

File: libs/category/src/models/generic-category.ts
Comment (xelaint) [inline libs/category/src/models/generic-category.ts:35]: `the category`?

File: libs/category/src/models/generic-category.ts
Comment (xelaint) [inline libs/category/src/models/generic-category.ts:39]: `the category`?

File: libs/category/src/models/generic-category.ts
Comment (xelaint) [inline libs/category/src/models/generic-category.ts:43]: `the category`?

File: libs/category/src/models/get-category-response.ts
Comment (xelaint) [inline libs/category/src/models/get-category-response.ts:25]: `the category`?

File: libs/category/state/src/facades/category-facade.interface.ts
Comment (xelaint) [inline libs/category/state/src/facades/category-facade.interface.ts:24]: `It exposes`?

File: libs/category/state/src/facades/category.facade.ts
Comment (xelaint) [inline libs/category/state/src/facades/category.facade.ts:32]: `It exposes`?

File: libs/category/state/src/selectors/category.selector.ts
Comment (xelaint) [inline libs/category/state/src/selectors/category.selector.ts:45]: Selects `the`

**Review by xelaint — CHANGES_REQUESTED** [review]

File: libs/category/src/models/category-breadcrumb.ts
Comment (lderrickable) [reply libs/category/src/models/category-breadcrumb.ts:19]: I meant `trie` but I think I'm misusing the term. I'll switch to "tree"

File: libs/category/src/models/generic-category.ts
Comment (lderrickable) [reply libs/category/src/models/generic-category.ts:27]: I'd rather use "this" when there are multiple categories "the" could be referring to. I could probably use "the" and presume that the reader will correctly assume I mean "this" (and they probably would), but I could also just use "this".

File: libs/category/state/src/selectors/category.selector.ts
Comment (lderrickable) [reply libs/category/state/src/selectors/category.selector.ts:45]: I'd rather use "a" here. "the" sounds weird to me. I had to sit here for 10 minutes trying to figure out why, but I think I've got it. I think "Selects a category" suggests a one of many relationship, whereas "Selects the category" does not. When you add the rest of it "Selects a category by id." vs "Selects the category by id." they do seem to mean the same thing, but it feels like the "the" article is working against the rest of the statement. Does that make sense to you, or am I crazy?

File: libs/category/state/src/facades/category.facade.ts
Comment (lderrickable) [reply libs/category/state/src/facades/category.facade.ts]: changed

File: libs/category/state/src/facades/category-facade.interface.ts
Comment (lderrickable) [reply libs/category/state/src/facades/category-facade.interface.ts]: changed

File: libs/category/src/models/get-category-response.ts
Comment (lderrickable) [reply libs/category/src/models/get-category-response.ts]: changed

File: libs/category/src/models/generic-category.ts
Comment (lderrickable) [reply libs/category/src/models/generic-category.ts]: changed

File: libs/category/src/models/generic-category.ts
Comment (lderrickable) [reply libs/category/src/models/generic-category.ts]: changed

File: libs/category/src/models/category-breadcrumb.ts
Comment (lderrickable) [reply libs/category/src/models/category-breadcrumb.ts]: changed

File: libs/category/src/models/category-breadcrumb.ts
Comment (lderrickable) [reply libs/category/src/models/category-breadcrumb.ts]: changed

File: libs/category/src/models/generic-category.ts
Comment (xelaint) [reply libs/category/src/models/generic-category.ts:27]: In that case, `a category` would probably make more sense? What is `this` category?

File: libs/category/state/src/selectors/category.selector.ts
Comment (xelaint) [inline libs/category/state/src/selectors/category.selector.ts:39]: Kind of related to this PR because of the docs you wrote — wouldn't it make more sense for this property to be named `selectCurrentCategory`.

File: libs/category/state/src/selectors/category.selector.ts
Comment (xelaint) [reply libs/category/state/src/selectors/category.selector.ts:45]: What if I phrased it to be selects the current category by id given that the other docs are referencing current category. selects a current category by id no longer makes sense.

File: libs/category/src/models/get-category-response.ts
Comment (griest024) [inline libs/category/src/models/get-category-response.ts:17]: I would say `The list of products that belong to the requested category`

File: libs/category/src/models/get-category-response.ts
Comment (griest024) [inline libs/category/src/models/get-category-response.ts:21]: I prefer `The requested category`

File: libs/category/state/src/reducers/category-entities/category-entities.reducer.ts
Comment (griest024) [inline libs/category/state/src/reducers/category-entities/category-entities.reducer.ts:20]: Say something about the reducer storing category entities in response to certain actions.

File: libs/category/state/src/selectors/category-page/category-page.selector.ts
Comment (griest024) [inline libs/category/state/src/selectors/category-page/category-page.selector.ts:64]: Remove double negative

**Review by griest024 — CHANGES_REQUESTED** [review]

File: libs/category/state/src/selectors/category.selector.ts
Comment (lderrickable) [reply libs/category/state/src/selectors/category.selector.ts:45]: If it was changed to a "current category" selector, then I would agree that "the" makes sense, because "the" designates a particular category from the beginning, i.e. the current category. Since there can only ever be one current category, "a current category" wouldn't make sense. But that's not what this selector does. It has no idea what the current category is. It just returns whatever category you want based on the id provided.

I think your argument is this: if a doc has a broad meaning (selects a category by id), then it should also be able to cover a more specific circumstance too (if I give it the id of the current category, it should return the current category). Therefore, the doc should cover the more specific circumstance of giving it a particular value. The doc should be rewritten as "Selects the category by id".

This argument sounds congruent to your argument: if a doc has a broad meaning (selects a category by id), then it should also be able to cover a more specific circumstance too (if I give it the id "4", it should return the category related to the id "4"). Therefore, the doc should cover the move specific circumstance of giving it a "4" id. The doc should be rewritten to "Selects the category with id=4."

Given, the second argument seems obviously wrong, because it takes the first argument to the extreme. But I think they are essentially the same. They both try to justify a more narrow doc by supplying a more narrow example that the selector covers

File: libs/category/src/models/generic-category.ts
Comment (lderrickable) [reply libs/category/src/models/generic-category.ts:27]: "This" category is the `DaffGenericCategory`. "a category" would make less sense, I think, because it would be even more arbitrary than "the".

Oh, that's just because I make mistakes sometimes (some might say, "often"). I'll change it to "this".

File: libs/category/state/src/selectors/category.selector.ts
Comment (lderrickable) [reply libs/category/state/src/selectors/category.selector.ts:39]: I think so, yea. I don't think I should make breaking changes in this PR though. I can remove the doc if you'd prefer.

File: libs/category/src/models/generic-category.ts
Comment (lderrickable) [reply libs/category/src/models/generic-category.ts:27]: Oh actually, nevermind, I meant to have that one be a "the". I chose "The number of products described by the category." because I know you prefer "the" to "this" and I didn't think "this" was necessary because there is no other category to get confused with as there is no reference to any other category in this particular description. So "the" category can only refer to the `DaffGenericCategory`.

File: libs/category/src/models/get-category-response.ts
Comment (lderrickable) [reply libs/category/src/models/get-category-response.ts]: changed

File: libs/category/src/models/get-category-response.ts
Comment (lderrickable) [reply libs/category/src/models/get-category-response.ts]: Changed

File: libs/category/state/src/reducers/category-entities/category-entities.reducer.ts
Comment (lderrickable) [reply libs/category/state/src/reducers/category-entities/category-entities.reducer.ts]: done

File: libs/category/state/src/selectors/category-page/category-page.selector.ts
Comment (lderrickable) [reply libs/category/state/src/selectors/category-page/category-page.selector.ts]: done

**Review by griest024 — DISMISSED** [review]

File: libs/category/state/src/selectors/category.selector.ts
Comment (xelaint) [reply libs/category/state/src/selectors/category.selector.ts:39]: I think you can keep the doc, but make an issue about updating the property name?

**Comment by xelaint** [comment]
There's two changes to `generic-category.ts` that still need to be addressed but lgtm after those changes + rebase.

File: libs/category/src/models/generic-category.ts
Comment (lderrickable) [reply libs/category/src/models/generic-category.ts:43]: This is the same situation as the other discussion we were having: 
> I'd rather use "this" when there are multiple categories "the" could be referring to. I could probably use "the" and presume that the reader will correctly assume I mean "this" (and they probably would), but I could also just use "this".

File: libs/category/src/models/generic-category.ts
Comment (lderrickable) [reply libs/category/src/models/generic-category.ts:51]: This is the same situation as the other discussion we were having:
> I'd rather use "this" when there are multiple categories "the" could be referring to. I could probably use "the" and presume that the reader will correctly assume I mean "this" (and they probably would), but I could also just use "this".

File: libs/category/state/src/selectors/category.selector.ts
Comment (lderrickable) [reply libs/category/state/src/selectors/category.selector.ts:39]: added: https://github.com/graycoreio/daffodil/issues/1648

**Review by xelaint — DISMISSED** [review]

**Comment by lderrickable** [comment]
I think this needs a reapproval by @griest024

**Review by griest024 — APPROVED** [review]

**Review by griest024 — APPROVED** [review]

**Pull request merged**
